hudi-hadoop-common/src/main/java/org/apache/hudi/io/hadoop/HoodieHBaseAvroHFileReader.java

```diff
@@ -338,32 +339,52 @@ public boolean hasNext() {
           return false;
         }
 
-        Cell c = Objects.requireNonNull(scanner.getCell());
-        byte[] keyBytes = copyKeyFromCell(c);
-        String key = new String(keyBytes);
-        // Check whether we're still reading records corresponding to the key-prefix
-        if (!key.startsWith(keyPrefix)) {
-          return false;
-        }
-
-        // Extract the byte value before releasing the lock since we cannot hold on to the returned cell afterwards
-        byte[] valueBytes = copyValueFromCell(c);
         try {
+          // Initialize scanner if needed
+          if (!initialized) {
+            initialized = true;
```

Comment:
we can eliminate the `initialized` variable, just use the `eof == true` to returns early for next `#hasNext` should be fine.